### PR TITLE
Update Vagrant to Ubuntu 14.04 and Ruby 2.1

### PR DESCRIPTION
**Pull request by edrex**
This change updates the Vagrant config to use Ubuntu 14.04 and Ruby 2.1 so that
the Vagrant environment works with the current version of Calagator. It was also tested briefly by merging into the *engine* branch.

- Replaces the Chef-based provisioning script with an equivalent shell script.
- Drops support for NFS-based shared directories and host gem cache directory, although
  that support can be restored if people want it.
- Adds PhantomJS as needed by the test suite.
- Doesn't include Sunspot, although it can be added if people want it.

Known issues: 

- Warnings when running `rake` and `rails` about missing X11 libs caused by the libnotify gem.
  This will happen in any Linux environment without X11 installed and should be addressed in a seperate issue.

Testing done:

- `rake` tests pass, except pending sunspot tests
- `rails server` runs, site appears to work

@reidab @aeschright @shawnacscott

### CODE REVIEW

**Comment by reidab** [comment]
Thanks! 

If it's not too much extra work, I'd love to see sunspot added to the Vagrant environment, as it can be one of the trickier things for new dev to get up and running. I'm also happy to merge this as-is and add a separate issue for adding sunspot — let me know.

**Comment by edrex** [comment]
Ok, added `openjdk-7-jre-headless` package which seems to be all that's needed for `bundle exec rake sunspot:solr:start` to work. We could add services for `sunspot:solr:start` and `RAILS_ENV=test sunspot:solr:start` but that would add ~100MB memory overhead per.

File: VAGRANT.md [hunk 1]
```diff
@@ -6,13 +6,7 @@ Using Vagrant for development
 Setup
 -----
 
-Setup Vagrant and its dependencies:
-
-1. Install Ruby: [http://www.ruby-lang.org/](http://www.ruby-lang.org/)
-3. Install VirtualBox: [http://www.virtualbox.org/](http://www.virtualbox.org/)
```
Comment (reidab) [inline VAGRANT.md]: I think people still need to install VirtualBox separately, since we're using it as the provider.

File: Vagrantfile [hunk 2]
```diff
@@ -5,59 +5,42 @@ if File.exist?(overrides)
...
+
+  # Share an additional folder to the guest VM. The first argument is
+  # the path on the host to the actual folder. The second argument is
+  # the path on the guest to mount the folder. And the optional third
+  # argument is a set of non-required options.
+  # config.vm.synced_folder "../data", "/vagrant_data"
+
+  # Use more memory so Bundler works.
+  config.vm.provider "virtualbox" do |vb|
+    vb.memory = "1024"
```
Comment (reidab) [inline Vagrantfile]: Per VAGRANT.md, this should pay attention to the MEMORY override if it's set.

**Comment by edrex** [comment]
Thanks for the review, updated.

**Comment by reidab** [comment]
Verified as working! Thanks!

I ran into an issue if Gemfile.local was present, but I think the same issue would have happened on our old Vagrant configuration.

**Pull request merged**